### PR TITLE
DT-72: Added retries to health check

### PR DESCRIPTION
**Pull request by SimonMitchellMOJ**

### CODE REVIEW

File: server/data/healthcheck.js [hunk 1]
```diff
@@ -1,41 +1,57 @@
...
+  maxFreeSockets: config.apis.elite2.agent.maxFreeSockets,
+  freeSocketTimeout: config.apis.elite2.agent.freeSocketTimeout,
+}
 
-          if (result.status === 200) {
-            return resolve('OK')
-          }
+function serviceCheckFactory(name, url) {
+  const keepaliveAgent = url.startsWith('https') ? new HttpsAgent(agentOptions) : new Agent(agentOptions)
+  logger.warn(`We have constructed a new service check here for ${url}`)
```
Comment (andrewrlee) [inline server/data/healthcheck.js]: meh.. was this just for debug during development? do we need to log this

File: server/data/healthcheck.js [hunk 2]
```diff
@@ -1,41 +1,57 @@
...
+          try {
+            if (error) {
+              logger.error(error.stack, `Error calling ${name}`)
+              reject(error)
+            } else if (result.status === 200) {
+              resolve('OK')
+            } else {
+              reject(result.status)
+            }
+          } catch (apiError) {
```
Comment (andrewrlee) [inline server/data/healthcheck.js]: not quite sure what would throw this error? the code in the try seems to only log or return a promise?

Comment (andrewrlee) [inline server/data/healthcheck.js]: when would this be thrown?

File: server/data/healthcheck.js (hunk 1 above)
Comment (SimonMitchellMOJ) [reply server/data/healthcheck.js]: Removed

File: server/data/healthcheck.js (hunk 2 above)
Comment (SimonMitchellMOJ) [reply server/data/healthcheck.js]: code removed

**Pull request merged**